File: app/src/main/java/ntktest/example/com/ntktest/data/RssSourceDao.java
package ntktest.example.com.ntktest.data;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

@Dao
public interface RssSourceDao {
    @Query("SELECT * FROM RssSource")
    List<RssSource> all();

    @Query("SELECT * FROM RssSource WHERE id = :id")
    RssSource getById(int id);

    @Insert
    void insert(RssSource... sources);

    @Update
    void update(RssSource... sources);

    @Delete
    void delete(RssSource... sources);
}

File: app/src/main/java/com/example/rssreader/data/RssSourceDao.kt
package com.example.rssreader.data

import androidx.room.*
import io.reactivex.Completable
import io.reactivex.Maybe
import io.reactivex.Single

@Dao
interface RssSourceDao {
    @Query("SELECT * FROM RssSource")
    fun all(): Single<List<RssSource>>

    @Query("SELECT * FROM RssSource WHERE id = :id")
    fun getById(id: Int): Maybe<RssSource>

    @Insert
    fun insert(vararg sources: RssSource): Completable

    @Update
    fun update(vararg sources: RssSource): Completable

    @Delete
    fun delete(vararg sources: RssSource): Completable
}